pkg/gke/create_test.go

```diff
@@ -0,0 +1,226 @@
+		Expect(managedCluster.Name).To(Equal(config.Spec.ClusterName))
+	})
+
+	It("should fail to create cluster", func() {
+		clusterServiceMock.EXPECT().
+			ClusterList(
+				ctx,
+				LocationRRN(config.Spec.ProjectID, Location(config.Spec.Region, config.Spec.Zone))).
+			Return(
+				&gkeapi.ListClustersResponse{
+					Clusters: []*gkeapi.Cluster{
+						{
+							Name: "test-cluster",
+						},
+					},
+				}, nil)
+
+		err := Create(ctx, clusterServiceMock, config)
+		Expect(err).To(HaveOccurred())
+	})
+
+})
+
+var _ = Describe("CreateNodePool", func() {
+	var (
+		mockController     *gomock.Controller
+		clusterServiceMock *mock_services.MockGKEClusterService
+		k8sVersion         = "1.25.12-gke.200"
+		clusterIpv4Cidr    = "10.42.0.0/16"
+		networkName        = "test-network"
+		subnetworkName     = "test-subnetwork"
+		emptyString        = ""
+		boolTrue           = true
+
+		nodePoolName      = "test-node-pool"
+		initialNodeCount  = int64(3)
+		maxPodsConstraint = int64(110)
+		nodePoolConfig    = &gkev1.GKENodePoolConfig{
+			Name:              &nodePoolName,
+			InitialNodeCount:  &initialNodeCount,
+			Version:           &k8sVersion,
+			MaxPodsConstraint: &maxPodsConstraint,
+			Config:            &gkev1.GKENodeConfig{},
+			Autoscaling: &gkev1.GKENodePoolAutoscaling{
+				Enabled:      true,
+				MinNodeCount: 3,
+				MaxNodeCount: 5,
+			},
+			Management: &gkev1.GKENodePoolManagement{
+				AutoRepair:  true,
+				AutoUpgrade: true,
+			},
+		}
+
+		config = &gkev1.GKEClusterConfig{
+			Spec: gkev1.GKEClusterConfigSpec{
+				Region:                "test-region",
+				ProjectID:             "test-project",
+				ClusterName:           "test-cluster",
+				Locations:             []string{""},
+				Labels:                map[string]string{"test": "test"},
+				ClusterIpv4CidrBlock:  &clusterIpv4Cidr,
+				KubernetesVersion:     &k8sVersion,
+				LoggingService:        &emptyString,
+				MonitoringService:     &emptyString,
+				EnableKubernetesAlpha: &boolTrue,
+				Network:               &networkName,
+				Subnetwork:            &subnetworkName,
+				NetworkPolicyEnabled:  &boolTrue,
+				MaintenanceWindow:     &emptyString,
+				IPAllocationPolicy: &gkev1.GKEIPAllocationPolicy{
+					UseIPAliases: true,
+				},
+				ClusterAddons: &gkev1.GKEClusterAddons{
+					HTTPLoadBalancing:        true,
+					NetworkPolicyConfig:      false,
+					HorizontalPodAutoscaling: true,
+				},
+				PrivateClusterConfig: &gkev1.GKEPrivateClusterConfig{
+					EnablePrivateEndpoint: false,
+					EnablePrivateNodes:    false,
+				},
+				MasterAuthorizedNetworksConfig: &gkev1.GKEMasterAuthorizedNetworksConfig{
+					Enabled: false,
+				},
+			},
+		}
+	)
+
+	BeforeEach(func() {
+		mockController = gomock.NewController(GinkgoT())
+		clusterServiceMock = mock_services.NewMockGKEClusterService(mockController)
+	})
+
+	AfterEach(func() {
+		mockController.Finish()
+	})
+
+	It("should successfully create cluster and node pool", func() {
```

Comment:
do we need a failing test case for node pool creation?

Reply:
fixed